app/src/main/java/com/lee989898/soptlee/signin/SignInActivity.kt

```diff
@@ -0,0 +1,47 @@
+package com.lee989898.soptlee.signin
+
+import android.app.Activity
+import android.content.Intent
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.widget.Toast
+import androidx.activity.result.contract.ActivityResultContracts
+import com.lee989898.soptlee.databinding.ActivitySignInBinding
+import com.lee989898.soptlee.main.HomeActivity
+import com.lee989898.soptlee.signup.SignUpActivity
+
+class SignInActivity : AppCompatActivity() {
+
+    private lateinit var binding: ActivitySignInBinding
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = ActivitySignInBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        val resultLauncher = registerForActivityResult(ActivityResultContracts.StartActivityForResult()){
+            result ->
+            if(result.resultCode == Activity.RESULT_OK){
+                val myData: Intent? = result.data
+                binding.signInIdEt.setText(myData!!.getStringExtra("id"))
+                binding.signInPasswordEt.setText(result.data?.getStringExtra("pwd"))
```

Comment:
```suggestion
                val myData = result.data ?: return@registerForActivityResult
                binding. signInIdEt.setText(myData.getStringExtra("id"))
                binding. signInPasswordEt.setText(myData.getStringExtra("pwd"))
```

이건 어떠신가요?
`binding.signInIdEt.setText(myData!!.getStringExtra("id"))` NPE 발생 소지가 있어보어요 👀

Reply:
아하 엘비스연산자로 null방지하면 좋겠네요!!

Reply:
```suggestion
                val myData: Intent? = result.data
                myData.let{
                    binding.signInIdEt.setText(myData.getStringExtra("id"))
                    binding.signInPasswordEt.setText(myData.getStringExtra("pwd"))
               }
```

kotlin에는 5개의 `scoped functions`가 있는데, 그 중 하나가 `let` 입니다.
이 let은 여러가지 상황에서 사용 가능한데, 그 중 하나가 null이 아닌 값으로만 코드 블록을 실행시키고 싶은 경우입니다.
위와 같이 코드를 작성한다면 myData가 null 이 아닌 경우에 대해서면, id와 pw 값을 설정해주는 식으로 사용 가능 하겠죠!
scope function 에 대해 조금 더 공부해보고 싶다면 이 [링크](https://kotlinlang.org/docs/scope-functions.html)를 참고하세요!

Reply:
헷갈렸는데 덕분에 이해했네요 감사합니다